### PR TITLE
Cleanup any Logstash PVCs after test

### LINKED ISSUE #7561: Logstash tests failing because volume can not be provisioned?
Since https://github.com/elastic/cloud-on-k8s/pull/7558 we have a new set of failures 

🐞 TestLogstashResolvingDollarVariableInStackMonitoring/Logstash_status_should_have_the_correct_status ~ aks
🐞 TestLogstashStackMonitoring/Logstash_status_should_have_the_correct_status ~ aks
🐞 TestLogstashResolvingDollarVariableInStackMonitoring/Logstash_status_should_have_the_correct_status ~ eks
🐞 TestLogstashStackMonitoring/Logstash_status_should_have_the_correct_status ~ eks
🐞 TestLogstashResolvingDollarVariableInStackMonitoring/All_expected_Pods_should_eventually_be_ready#01 ~ gke
🐞 TestLogstashStackMonitoring/Logstash_status_should_have_the_correct_status ~ gke
🐞 TestLogstashResolvingDollarVariableInStackMonitoring/All_expected_Pods_should_eventually_be_ready ~ resilience
🐞 TestLogstashResolvingDollarVariableInStackMonitoring/All_expected_Pods_should_eventually_be_ready#01 ~ stack-8-13-0-sna
🐞 TestLogstashStackMonitoring/Logstash_status_should_have_the_correct_status ~ stack-8-13-0-sna
🐞 TestLogstashResolvingDollarVariableInStackMonitoring/All_expected_Pods_should_eventually_be_ready#01 ~ tanzu
🐞 TestLogstashStackMonitoring/Logstash_status_should_have_the_correct_status ~ tanzu

```
pod test-ls-mon-logs-mz7q-es-masterdata-1 is not Ready.
        	            	Status:{
        	            	    "phase": "Pending",
        	            	    "conditions": [
        	            	        {
        	            	            "type": "PodScheduled",
        	            	            "status": "False",
        	            	            "lastProbeTime": null,
        	            	            "lastTransitionTime": "2024-02-15T06:44:54Z",
        	            	            "reason": "Unschedulable",
        	            	            "message": "0/3 nodes are available: 3 node(s) didn't find available persistent volumes to bind. preemption: 0/3 nodes are available: 3 Preemption is not helpful for scheduling."
        	            	        }
        	            	    ],
```

Looking at `455-TestLogstashResolvingDollarVariableInStackMonitoring-Logstash_status_should_have_the_correct_status`

```
cat pods.json | jq -r '.items[] | select(.status.phase == "Running") | .metadata.name'
test-ls-mon-logs-stb7-es-masterdata-0
test-ls-mon-logs-stb7-es-masterdata-1
test-ls-mon-metrics-46xd-es-masterdata-0
test-ls-mon-metrics-46xd-es-masterdata-1
```

```
cat persistentvolumeclaims.json | jq '.items[] | "\(.metadata.name): \(.spec.storageClassName) \(.spec.resources.requests.storage)"'
"elasticsearch-data-test-ls-mon-logs-stb7-es-masterdata-0: e2e-default 1Gi"
"elasticsearch-data-test-ls-mon-logs-stb7-es-masterdata-1: e2e-default 1Gi"
"elasticsearch-data-test-ls-mon-metrics-46xd-es-masterdata-0: e2e-default 1Gi"
"elasticsearch-data-test-ls-mon-metrics-46xd-es-masterdata-1: e2e-default 1Gi"
"logstash-data-d-2nmh-ls-0: e2e-default 1Gi"
"logstash-data-global-ca-f42w-ls-0: e2e-default 1536Mi"
"logstash-data-logstash-sample-2b49-ls-0: e2e-default 1536Mi"
"logstash-data-logstash-sample-4j86-ls-0: e2e-default 1536Mi"
"logstash-data-logstash-sample-4j86-ls-1: e2e-default 1536Mi"
"logstash-data-logstash-sample-4rcp-ls-0: e2e-default 1536Mi"
"logstash-data-logstash-sample-4rcp-ls-1: e2e-default 1536Mi"
"logstash-data-logstash-sample-4rcp-ls-2: e2e-default 1536Mi"
"logstash-data-logstash-sample-5hmq-ls-0: e2e-default 1536Mi"
"logstash-data-test-env-logstash-hlhm-ls-0: e2e-default 1536Mi"
"logstash-data-test-keystore-with-default-pw-hfcm-ls-0: e2e-default 1536Mi"
"logstash-data-test-keystore-with-default-pw-hx2f-ls-0: e2e-default 1536Mi"
"logstash-data-test-ls-es-out-mtrm-ls-0: e2e-default 1536Mi"
"logstash-data-test-ls-mon-a-9424-ls-0: e2e-default 1536Mi"
"logstash-data-test-ls-mon-dollar-h72s-ls-0: e2e-default 1536Mi"
"logstash-data-test-ls-reload-2czx-ls-0: e2e-default 1536Mi"
"logstash-data-test-ls-version-upgrade-8x-xdj8-ls-0: e2e-default 1536Mi"
"logstash-data-test-ls-version-upgrade-8x-xdj8-ls-1: e2e-default 1536Mi"
"logstash-data-test-multiple-custom-logstash-fkrq-ls-0: e2e-default 1536Mi"
"logstash-data-test-multiple-custom-logstash-jfs4-ls-0: e2e-default 1536Mi"
"logstash-data-test-multiple-custom-logstash-x85h-ls-0: e2e-default 1536Mi"
"logstash-data-test-multiple-logstashes-k566-ls-0: e2e-default 1536Mi"
"logstash-data-test-multiple-logstashes-k566-ls-1: e2e-default 1536Mi"
"logstash-data-test-multiple-logstashes-k566-ls-2: e2e-default 1536Mi"
"logstash-data-test-pipeline-ref-xcq5-ls-0: e2e-default 1536Mi"
"logstash-data-test-single-logstash-zkmc-ls-0: e2e-default 1536Mi"
"logstash-data-test-split-pipeline-mwwg-ls-0: e2e-default 1536Mi"
"pq-d-2nmh-ls-0: e2e-default 2Gi"
```
I think we are just exhausting the available local storage because IIRC Logstash volumes are not deleted after the test (to be verified)

### PR DESCRIPTION
**Pull request by pebrc**
Fixes #7561 

Try to delete Logstash PVCs after test.

### CODE REVIEW

**Comment by pebrc** [comment]
buildkite test this -f p=gke,E2E_TAGS=logstash

File: test/e2e/test/logstash/steps.go [hunk 1]
```diff
@@ -145,5 +147,15 @@ func (b Builder) DeletionTestSteps(k *test.K8sClient) test.StepList {
 				return k.CheckPodCount(0, b.ListOptions()...)
 			}),
 		},
+		{
+			Name: "Cleanup any persistent volumes belonging to Logstash",
+			Test: test.Eventually(func() error {
+				if err := k.Client.DeleteAllOf(context.Background(), &corev1.PersistentVolumeClaim{},
+					client.MatchingLabels{lslabels.NameLabelName: b.Logstash.Name}); err != nil && !apierrors.IsNotFound(err) {
```
Comment (thbkrkr) [inline test/e2e/test/logstash/steps.go:154]: I think you need to specify a namespace otherwise the request will be refused.

Comment (pebrc) [reply test/e2e/test/logstash/steps.go]: Thanks @thbkrkr I did not think of that.

**Comment by pebrc** [comment]
buildkite test this -f p=gke,E2E_TAGS=logstash

**Review by kvalliyurnatt — APPROVED** [review]

**Pull request merged**